tests/run-make/short-ice/rmake.rs

```diff
@@ -0,0 +1,31 @@
+// Backtraces in internal compiler errors used to be unbearably long, spanning
+// multiple hundreds of lines. A fix was pushed in #108938, and this test gathers
+// varied metrics on level 1 and full-level backtraces to check that the output
+// was shortened down to an appropriate length.
+// See https://github.com/rust-lang/rust/issues/107910
+
+use run_make_support::rustc;
+use std::env;
+
+fn main() {
+    env::set_var("RUST_BACKTRACE", "1");
+    let mut rust_test_1 = rustc().input("src/lib.rs").arg("-Ztreat-err-as-bug=1").run_fail();
```

Comment:
Suggestion: this could be set on the `Rustc` command, and not set for the run-make process then inherited in child processes. Same for the other `set_var`.

Reply:
Alright, I added a new helper function which should hopefully do the trick! @rustbot review

Reply:
Oh what I had in mind was just using `Command::env` because I don't think setting backtrace is too common. But we can keep the helper if you think that is more readable.

Reply:
I do think that it's more readable - it shows that it's precisely for that specific Rustc call, and it's a tool for future test writers.